Reorganization of project structure

JakeDawkins opened the pull request:
This is a work in progress, and right now _everything_ is broken. 

I've moved a good bit of the directories, but I'm still confused about some of the `components` folder items and the parent level items. None of the imports have been fixed. 

I also added a couple lines to the readme, but I think it would be helpful to fully explain the structure in a Medium post or something later on and link to here.

jbaxleyiii commented:
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://a3c403a7-3133-44d3-b6ea-89199b6593f1.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/NewSpring/Holtzman/branch/re-org/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

JakeDawkins commented:
I believe this is done, @jbaxleyiii 

This PR has:
- File structure changes
- Dropping of the `series.SingleItem` style file naming on non-deprecated files
- abstraction of the progress bar used in giving dashboards
- added a couple index files for folders with collections of components (like buttons, cards)
- removed READMEs that had nothing useful in them
- changed some strange naming (`switch > toggle` and `toggle > tabs`) to be clearer
- This still uses relative imports, as I had problems getting both jest watch mode and `jest file.js` to work. 
- and more 🍻

jbaxleyiii commented:
@JakeDawkins a couple merge conflicts to clean up. After QA of #1697 and #1696 tomorrow morning, I'll cut a build and this can be merged 💪
